.github/workflows/generate-config.yaml

```diff
@@ -80,61 +91,85 @@ jobs:
+          IFS=',' read -ra NETWORKS <<< "${{ env.NETWORKS }}"
+          for NETWORK in "${NETWORKS[@]}"; do
             echo "Generating config for $NETWORK"
             poetry run python generate.py --network "$NETWORK" gen-configs all-configs
           done
 
           git add --all
-          git commit -m "create branch ${{ steps.branch-name.outputs.branch }}"
+          git commit -m "chore: update generated configuration files for ${{ env.NETWORKS }}"
           git push --set-upstream origin ${{ steps.branch-name.outputs.branch }}
 
       - name: Create Pull Request
         id: create_pr
         run: |
-          cd ${{ secrets.CONFIG_REPO }}
+          cd config-repo
           PR_TITLE="chore: update configs for ${{ env.NETWORKS }} from axelar-contract-deployments"
           PR_BODY="Auto-generated PR with updated configuration files"
           PR_HEAD="${{ steps.branch-name.outputs.branch }}"
           PR_BASE="main"
-          RESPONSE=$(curl -s -X POST \
+          # Capture both response and HTTP status code
+          RESPONSE=$(curl -s -w "%{http_code}" -o response.json -X POST \
             -H "Authorization: token ${{ secrets.CHAINS_REPO_CICD_TOKEN }}" \
             -H "Accept: application/vnd.github.v3+json" \
             https://api.github.com/repos/axelarnetwork/${{ secrets.CONFIG_REPO }}/pulls \
             -d "{\"title\":\"$PR_TITLE\",\"body\":\"$PR_BODY\",\"head\":\"$PR_HEAD\",\"base\":\"$PR_BASE\"}")
-          PR_NUMBER=$(echo $RESPONSE | jq -r .number)
+          HTTP_STATUS=$(tail -c 3 <<< "$RESPONSE")
+          cat response.json
+          PR_NUMBER=$(jq -r .number response.json)
+          if [[ "$HTTP_STATUS" != "201" ]] || [[ "$PR_NUMBER" == "null" || -z "$PR_NUMBER" ]]; then
+            echo "Failed to create PR. HTTP status: $HTTP_STATUS, PR number: $PR_NUMBER"
+            cat response.json
+            exit 1
+          fi
           echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
 
       - name: Merge Pull Request
+        if: steps.create_pr.outcome == 'success'
         run: |
+          cd config-repo
           PR_NUMBER=${{ env.PR_NUMBER }}
-          PR_TITLE="chore: update configurations for ${{ env.NETWORKS }} from axelar-contract-deployments"
-          curl -s -X PUT \
+          PR_TITLE="chore: update configs for ${{ env.NETWORKS }} from axelar-contract-deployments"
+          # Attempt to merge the PR and capture both response and HTTP status code
+          RESPONSE=$(curl -s -w "%{http_code}" -o merge_response.json -X PUT \
             -H "Authorization: token ${{ secrets.CHAINS_REPO_CICD_TOKEN }}" \
             -H "Accept: application/vnd.github.v3+json" \
             https://api.github.com/repos/axelarnetwork/${{ secrets.CONFIG_REPO }}/pulls/$PR_NUMBER/merge \
-            -d "{\"commit_title\":\"Merge PR #$PR_NUMBER: $PR_TITLE\",\"merge_method\":\"squash\"}"
+            -d "{\"commit_title\":\"Merge PR #$PR_NUMBER: $PR_TITLE\",\"merge_method\":\"squash\"}")
```

Comment:
can curl just fail/return non 0 exit code if the cmd fails instead of needing to parse the response?

Reply:
Ok, please update axelar-configs repo action to handle the response status correctly as well